spec/helpers/perf/drivers/docker.lua

```diff
@@ -225,81 +227,102 @@ function _M:start_upstreams(conf, port_count)
   return uris
 end
 
-function _M:start_kong(version, kong_conf)
+local function id_from_kong_conf(kong_conf)
+  local id = 'conf:'
+  for k, v in pairs(kong_conf) do
+    id = id .. k .. ':' .. v .. ';'
+  end
+  return id
+end
+
+function _M:_hydrate_kong_configuration(kong_conf, driver_conf)
+  local config = ''
+  for k, v in pairs(kong_conf) do
+    config = string.format("%s -e KONG_%s=%s", config, k:upper(), v)
+  end
+  config = config .. " -e KONG_PROXY_ACCESS_LOG=/dev/null -p 8001 -e KONG_ADMIN_LISTEN=0.0.0.0:8001 "
+
+  -- adds database configuration
+  if kong_conf['database'] == nil then
+    config = config .. " --link " .. self.psql_ct_id .. ":postgres " ..
+    "-e KONG_PG_HOST=postgres " ..
+    "-e KONG_PG_DATABASE=kong_tests "
+  end
+
+  if driver_conf['dns'] ~= nil then
+    for name, address in pairs(driver_conf['dns']) do
+      config = string.format("%s --link %s:%s", config, address, name)
+    end
+  end
+
+  return config
+end
+
+function _M:start_kong(version, kong_conf, driver_conf)
   if not version then
     error("Kong version is not defined", 2)
   end
 
   local use_git
   local image = "kong"
+  local kong_conf_id = id_from_kong_conf(kong_conf)
```

Comment:
as discussed offline, this function will need to be changed to a determinisitc ID (preferbaly from driver_conf) to avoid instances are left orphaned in start_kong.